src/carousel/carousel.scss

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@import '../style/index';
+@import '../text/index';
+@import '../button/button';
+
+@mixin iui-carousel {
+  @include iui-reset;
+  display: flex;
+  flex-direction: column;
+  position: relative;
+}
+
+@mixin iui-carousel-slider {
+  display: flex;
+  position: relative;
+  overflow: hidden;
+  list-style: none;
+  margin: 0;
+  padding: 0;
+
+  &-item {
+    position: absolute;
+    margin-left: auto;
+    margin-right: auto;
+    left: 0;
+    right: 0;
+    flex-shrink: 0;
+    width: 100%;
+    box-sizing: border-box;
+    @media (prefers-reduced-motion: no-preference) {
+      transition: transform $iui-speed-slow ease-in-out;
+    }
+
+    &.iui-previous {
+      transform: translateX(-100%);
+    }
+
+    &.iui-next {
+      transform: translateX(100%);
+    }
+
+    &.iui-current {
+      transform: translateX(0);
+    }
+
+    &-image {
+      width: 100%;
+      height: 100%;
+      background-position: center;
+      background-size: cover;
+    }
+
+    &-title {
+      position: absolute;
+      left: 0;
+      bottom: 0;
+      padding: $iui-baseline $iui-sm;
+      box-sizing: border-box;
+      width: 100%;
+      transition: background-color $iui-speed-fast ease;
+      @include themed {
+        background-color: rgba(0, 0, 0, t(iui-opacity-4));
+        color: t(iui-color-foreground-accessory);
+      }
+
+      &:hover {
+        @include themed {
+          background-color: rgba(0, 0, 0, t(iui-opacity-3));
+        }
+      }
+    }
+  }
+
+  + .iui-carousel-navigation {
+    @include themed {
+      border-top: $iui-xxs solid t(iui-color-background-4);
+    }
+  }
+}
+
+@mixin iui-carousel-navigation {
+  display: flex;
+  align-items: center;
+  height: $iui-baseline * 3;
+
+  &-dots {
+    @include iui-reset;
+    display: flex;
+    flex-grow: 1;
+    white-space: nowrap;
+    overflow: hidden;
+    justify-content: center;
+    align-items: center;
+    list-style: none;
+  }
+
+  &-dot {
+    @include iui-button;
```

Comment:
Sure we need all of the rules of button if everything seems to be set with the `:after` rule?

Reply:
I used the button mixin so that the dots width & height would match the left / right / pause buttons.  This way if we ever decided to change the button sizing, the dots would automatically change to match.  The `::after` is only used to place the dot.

Reply:
So if we care only about sizing, shouldnt mixin `iui-button-size($size: small)` be enough?

Reply:
If I remove `@include iui-button;` we get something like this:
![Screen Shot 2022-01-24 at 2 48 38 PM](https://user-images.githubusercontent.com/849817/150853781-500e226b-4e83-4a82-b9e7-ab0498203bbf.png)

So I need to add:
```scss
border: none;
cursor: pointer;
```

The button mixin included a 1px transparent border which serves us no purpose here, so test images are changing ever so slightly on my last push.